kv/kvserver: TestReplicaRangefeedPushesTransactions failed

kv/kvserver.TestReplicaRangefeedPushesTransactions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9419176?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9419176?buildTab=artifacts#/) on master @ [07c7d4b3abfa1dc79d8fda385132a27c81665fee](https://github.com/cockroachdb/cockroach/commits/07c7d4b3abfa1dc79d8fda385132a27c81665fee):


```
=== RUN   TestReplicaRangefeedPushesTransactions
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaRangefeedPushesTransactions564631963
    test_log_scope.go:79: use -show-logs to present logs inline
    replica_rangefeed_test.go:1212: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/replica_rangefeed_test.go:1212
        	Error:      	Expect "pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=d2d4cd4d key=/Table/106/1/1/0 iso=Serializable pri=0.04500505 epo=0 ts=1680599766.098914290,0 min=1680599766.098914290,0 seq=1} lock=true stat=PENDING rts=1680599766.098914290,0 wto=false gul=1680599766.598914290,0" to match "TransactionRetryError: retry txn"
        	Test:       	TestReplicaRangefeedPushesTransactions
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaRangefeedPushesTransactions564631963
--- FAIL: TestReplicaRangefeedPushesTransactions (38.02s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/replication
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicaRangefeedPushesTransactions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-26500

kv/kvserver.TestReplicaRangefeedPushesTransactions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9444084?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9444084?buildTab=artifacts#/) on master @ [ff2c4a3454a63bd74fe9a2feed53a7bdcfaf5191](https://github.com/cockroachdb/cockroach/commits/ff2c4a3454a63bd74fe9a2feed53a7bdcfaf5191):


```
=== RUN   TestReplicaRangefeedPushesTransactions
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaRangefeedPushesTransactions3835198380
    test_log_scope.go:79: use -show-logs to present logs inline
    replica_rangefeed_test.go:1212: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/replica_rangefeed_test.go:1212
        	Error:      	Expect "pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=6475f58a key=/Table/106/1/1/0 iso=Serializable pri=0.03217706 epo=0 ts=1680690201.757710149,0 min=1680690201.757710149,0 seq=1} lock=true stat=PENDING rts=1680690201.757710149,0 wto=false gul=1680690202.257710149,0" to match "TransactionRetryError: retry txn"
        	Test:       	TestReplicaRangefeedPushesTransactions
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaRangefeedPushesTransactions3835198380
--- FAIL: TestReplicaRangefeedPushesTransactions (39.19s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicaRangefeedPushesTransactions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicaRangefeedPushesTransactions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9509032?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9509032?buildTab=artifacts#/) on master @ [85e41ca8d3d9edacf5ee3061a2591b159b9b0502](https://github.com/cockroachdb/cockroach/commits/85e41ca8d3d9edacf5ee3061a2591b159b9b0502):


```
=== RUN   TestReplicaRangefeedPushesTransactions
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaRangefeedPushesTransactions1080303313
    test_log_scope.go:79: use -show-logs to present logs inline
    replica_rangefeed_test.go:1212: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/replica_rangefeed_test.go:1212
        	Error:      	Expect "pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=66b31c98 key=/Table/106/1/1/0 iso=Serializable pri=0.04029488 epo=0 ts=1681033858.818356751,0 min=1681033858.818356751,0 seq=1} lock=true stat=PENDING rts=1681033858.818356751,0 wto=false gul=1681033859.318356751,0" to match "TransactionRetryError: retry txn"
        	Test:       	TestReplicaRangefeedPushesTransactions
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaRangefeedPushesTransactions1080303313
--- FAIL: TestReplicaRangefeedPushesTransactions (41.16s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicaRangefeedPushesTransactions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicaRangefeedPushesTransactions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9518046?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9518046?buildTab=artifacts#/) on master @ [85e41ca8d3d9edacf5ee3061a2591b159b9b0502](https://github.com/cockroachdb/cockroach/commits/85e41ca8d3d9edacf5ee3061a2591b159b9b0502):


```
=== RUN   TestReplicaRangefeedPushesTransactions
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaRangefeedPushesTransactions261928654
    test_log_scope.go:79: use -show-logs to present logs inline
    replica_rangefeed_test.go:1212: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/replica_rangefeed_test.go:1212
        	Error:      	Expect "pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=830f5d5d key=/Table/106/1/1/0 iso=Serializable pri=0.01898143 epo=0 ts=1681120641.836144977,0 min=1681120641.836144977,0 seq=1} lock=true stat=PENDING rts=1681120641.836144977,0 wto=false gul=1681120642.336144977,0" to match "TransactionRetryError: retry txn"
        	Test:       	TestReplicaRangefeedPushesTransactions
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaRangefeedPushesTransactions261928654
--- FAIL: TestReplicaRangefeedPushesTransactions (39.31s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicaRangefeedPushesTransactions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicaRangefeedPushesTransactions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9539847?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9539847?buildTab=artifacts#/) on master @ [a5d61d2cf2bfc11bd6bdc27dcab32467334fb123](https://github.com/cockroachdb/cockroach/commits/a5d61d2cf2bfc11bd6bdc27dcab32467334fb123):


```
=== RUN   TestReplicaRangefeedPushesTransactions
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaRangefeedPushesTransactions1212862857
    test_log_scope.go:79: use -show-logs to present logs inline
    replica_rangefeed_test.go:1212: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/replica_rangefeed_test.go:1212
        	Error:      	Expect "pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=f47f4a39 key=/Table/106/1/1/0 iso=Serializable pri=0.01100902 epo=0 ts=1681197342.438773657,0 min=1681197342.438773657,0 seq=1} lock=true stat=PENDING rts=1681197342.438773657,0 wto=false gul=1681197342.938773657,0" to match "TransactionRetryError: retry txn"
        	Test:       	TestReplicaRangefeedPushesTransactions
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaRangefeedPushesTransactions1212862857
--- FAIL: TestReplicaRangefeedPushesTransactions (38.62s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicaRangefeedPushesTransactions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicaRangefeedPushesTransactions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9556176?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9556176?buildTab=artifacts#/) on master @ [582880ae2bb5d08e83a9871735deeeb3ed892e7d](https://github.com/cockroachdb/cockroach/commits/582880ae2bb5d08e83a9871735deeeb3ed892e7d):


```
=== RUN   TestReplicaRangefeedPushesTransactions
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaRangefeedPushesTransactions4170283496
    test_log_scope.go:79: use -show-logs to present logs inline
    replica_rangefeed_test.go:1212: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/replica_rangefeed_test.go:1212
        	Error:      	Expect "pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=773fdbf2 key=/Table/106/1/1/0 iso=Serializable pri=0.00249119 epo=0 ts=1681296527.765918154,0 min=1681296527.765918154,0 seq=1} lock=true stat=PENDING rts=1681296527.765918154,0 wto=false gul=1681296528.265918154,0" to match "TransactionRetryError: retry txn"
        	Test:       	TestReplicaRangefeedPushesTransactions
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaRangefeedPushesTransactions4170283496
--- FAIL: TestReplicaRangefeedPushesTransactions (39.31s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicaRangefeedPushesTransactions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/replica_rangefeed_test.go
+++ b/pkg/kv/kvserver/replica_rangefeed_test.go
@@ -1209,7 +1209,7 @@ func TestReplicaRangefeedPushesTransactions(t *testing.T) {
 
 	// The txn should not be able to commit since its commit timestamp was pushed
 	// and it has observed its timestamp.
-	require.Regexp(t, "TransactionRetryError: retry txn", tx1.Commit())
+	require.Regexp(t, "TransactionRetryWithProtoRefreshError", tx1.Commit())
 
 	// Make sure the RangeFeed hasn't errored yet.
 	select {

